Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -1709,6 +1713,68 @@ NiftiImageIO::SetImageIOOrientationFromNIfTI(unsigned short int dims)
     {
       m_Origin[2] = 0.0;
     }
+
+    if (this->m_NiftiImage->nifti_type == 0)
+    { // only do this for Analyze file format
+      SpatialOrientationAdapter::DirectionType   dir;
+      SpatialOrientationAdapter::OrientationType orient;
+      switch (this->m_NiftiImage->analyze75_orient)
+      {
+        case a75_transverse_unflipped:
+          orient = SpatialOrientation::ITK_COORDINATE_ORIENTATION_RPI;
+          break;
+        case a75_sagittal_unflipped:
+          orient = SpatialOrientation::ITK_COORDINATE_ORIENTATION_PIR;
+          break;
+        case a75_coronal_unflipped:
+          orient = SpatialOrientation::ITK_COORDINATE_ORIENTATION_RIP;
+          break;
+        case a75_transverse_flipped:
+          orient = SpatialOrientation::ITK_COORDINATE_ORIENTATION_RAI;
+          break;
+        case a75_sagittal_flipped:
+          orient = SpatialOrientation::ITK_COORDINATE_ORIENTATION_PIL;
+          break;
+        case a75_coronal_flipped:
+          orient = SpatialOrientation::ITK_COORDINATE_ORIENTATION_RSP;
+          break;
+        case a75_orient_unknown:
+          orient = SpatialOrientation::ITK_COORDINATE_ORIENTATION_RIP;
+          break;
+      }
+      dir = OrientAdapterType().ToDirectionCosines(orient);
+      const int           max_defined_orientation_dims = (dims > 3) ? 3 : dims;
+      std::vector<double> xDirection(dims, 0.0);
```

Comment:
This and below lines could/should be re-written as a `for d=0..2` loop.

Reply:
simplified